typed-racket-test/succeed/with-integer-linear-arith2.rkt

```diff
@@ -16,7 +16,7 @@
 (ann (vector 0 1 2 3 4 5 6 7) (V8 Any))
 (ann (vector 0 1 2 3 4 5 6 7) (V8 Byte))
 (ann '#(0 1 2 3 4 5 6 7) (V8 Any))
-(ann '#(0 1 2 3 4 5 6 7) (V8 Byte))
+(ann '#(0 1 2 3 4 5 6 7) (Refine [v : (Immutable-Vector Byte Byte Byte Byte Byte Byte Byte Byte)] (= 8 (vector-length v))))
```

Comment:
@pnwamk without this change, I get an error like this:

```
foo.rkt:7:5: Type Checker: type mismatch
  expected: (Refine
             (d : (U (Immutable-Vectorof Byte) (Mutable-Vectorof Byte)))
             (= 8 (vector-length d)))
  given: (Refine
          (e
           :
           (Immutable-Vector
            Integer
            Integer
            Integer
            Integer
            Integer
            Integer
            Integer
            Integer))
          (= (vector-length e) 8))
  in: (quote #(0 1 2 3 4 5 6 7))
```

Any idea how I could fix the error without changing this test?

IMO the "given" should be a subtype of the "expected" ... but I don't know how to change Typed Racket's opinion.